airflow/models/taskinstance.py

```diff
@@ -2095,8 +2096,7 @@ def get_email_subject_content(self, exception: BaseException) -> Tuple[str, str,
             html_content_err = jinja_env.from_string(default_html_content_err).render(**jinja_context)
 
         else:
-            jinja_context = self.get_template_context()
-            jinja_context.update(additional_context)
+            jinja_context = {**cast(dict, self.get_template_context()), **additional_context}
```

Comment:
This is where the deprecations are from, I’ll fix it.

Reply:
Yeah, I'm on it already @uranusjr

Reply:
(I've got another case of this inside utils.helpers too0